core/commonTypes.ml

```diff
@@ -131,6 +129,33 @@ let pk_type     = PrimaryKind.Type
 let pk_row      = PrimaryKind.Row
 let pk_presence = PrimaryKind.Presence
 
+module Kind = struct
+  type t = PrimaryKind.t * Subkind.t
+    [@@deriving eq,show]
+end
+
+module Quantifier = struct
+  type t = int * Kind.t
+    [@@deriving show]
+
+  let to_var = function
```

Comment:
I'd be inclined to call these `get_var`, `get_kind`, ... or just `var` or `kind`, since to me `to_*` connotes translation to an equivalent (e.g. serialized) format rather than projecting/getting a part of a data structure.  Alternatively, it seems like `Quantifier.t` is mostly being used as a record `{var:int, kind: Kind.t}`, where `Kind.t` itself is a record `{primary:PrimaryKind.t, subkind: Subkind.t}`.  Defining them this way would obviate the need for all of these projection functions, and allow factoring `to_primary_kind` as `.kind.primary` etc.

Reply:
Indeed, good points.

>  to me to_* connotes translation to an equivalent (e.g. serialized) format rather than projecting/getting a part of a data structure.

I feel similarly. However, in the previous refactorings we have followed the `to_` convention (c.f. `Binder.to_name`, `Binder.to_type`) and I wanted to stay consistent.

Reply:
OK, I hadn't seen that.  Maybe agreeing on recordifying more things would be good (cf. #533 )

I still think `to_*` is potentially confusing since it suggests an information-preserving conversion or serialization (dual to `from_*`) rather than a projection.  But not a blocker to merging.